### PR TITLE
[FEAT]: Poly API bindings for golang

### PR DESCRIPTION
**Pull request by jeremyfelder**
## Describe the changes

This PR add golang bindings for the Poly API.

In order to support running tests for both ntt and poly APis, tests have been moved to their own package under each curve/field. This allows us to run all the tests in between a curve/field specific setup and teardown.

TODO:
- [ ] Add DeviceSlice views into Polys

### CODE REVIEW

File: wrappers/golang/internal/generator/config/bls12377.go [hunk 1]
```diff
@@ -5,7 +5,7 @@ func init() {
 		PackageName:         "bls12377",
 		Curve:               "bls12_377",
 		GnarkImport:         "bls12-377",
-		SupportsPoseidon:    true,
```
Comment (ChickenLover) [inline wrappers/golang/internal/generator/config/bls12377.go:8]: I think you accidentally removed poseidon

File: wrappers/golang/internal/generator/config/grumpkin.go [hunk 2]
```diff
@@ -5,6 +5,7 @@ func init() {
 		PackageName:         "grumpkin",
 		Curve:               "grumpkin",
 		GnarkImport:         "",
+		SupportsPoly:        false,
```
Comment (ChickenLover) [inline wrappers/golang/internal/generator/config/grumpkin.go:8]: but why?

File: wrappers/golang/internal/generator/ntt/generate.go [hunk 3]
```diff
@@ -13,12 +13,6 @@ var nttTemplates = map[string]string{
 	"header":       "ntt/templates/ntt.h.tmpl",
 }
 
-var ecnttTemplates = map[string]string{
```
Comment (ChickenLover) [inline wrappers/golang/internal/generator/ntt/generate.go:16]: What happened here? Are they not needed anymore?

Comment (jeremyfelder) [reply wrappers/golang/internal/generator/ntt/generate.go:16]: it was a left over that wasn't removed. Ecntt generator was moved to its own folder

**Review by LeonHibnik — APPROVED** [review]

**Pull request merged**